`TextChoices` subclasses declared inside a class cause "Unable to resolve type of choices property" error

# Bug report

## What's wrong

Having an enum that extends `TextChoices` declared inside another class (e.g. a model), results in the following error when running `mypy`:
```
file.py: error: Unable to resolve type of choices property  [misc]
```

Here's an example of such definition:
```python

class MyModel:
    class MyEnum(models.TextChoices):
        foo = "foo", "Foo"
        bar = "bar", "Bar"

a = MyEnum.foo
```

Running mypy against that:
```
file.py:9: error: Unable to resolve type of choices property  [misc]
```

## How is that should be

No error should happen.

## System information

- OS:
- `python` version: 3.12.9
- `django` version: 5.1.8
- `mypy` version: 1.13.0
- `django-stubs` version: 5.2.0
- `django-stubs-ext` version: 5.2.0

the `master`  branch @ e566001 seems to be failing for me (both on mypy 1.13.0 and 1.15.0):
```
$ mypy file.py --show-traceback
.../lib/python3.12/site-packages/django-stubs/http/request.pyi:56: error: INTERNAL ERROR -- Please try using mypy master on GitHub:
https://mypy.readthedocs.io/en/stable/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 1.15.0
Traceback (most recent call last):
  File "mypy/semanal.py", line 7240, in accept
  File "mypy/nodes.py", line 1345, in accept
  File "mypy/semanal.py", line 3206, in visit_assignment_stmt
  File "mypy/semanal.py", line 3733, in process_type_annotation
  File "mypy/semanal.py", line 6894, in defer
AssertionError: Must not defer during final iteration
.../lib/python3.12/site-packages/django-stubs/http/request.pyi:56: : note: use --pdb to drop into pdb

```

> the `master` branch @ [e566001](https://github.com/typeddjango/django-stubs/commit/e5660012c2d327bca2446d9e6c6b3033b13e517c) seems to be failing for me (both on mypy 1.13.0 and 1.15.0):
> 
> ```
> $ mypy file.py --show-traceback
> .../lib/python3.12/site-packages/django-stubs/http/request.pyi:56: error: INTERNAL ERROR -- Please try using mypy master on GitHub:
> https://mypy.readthedocs.io/en/stable/common_issues.html#using-a-development-mypy-build
> Please report a bug at https://github.com/python/mypy/issues
> version: 1.15.0
> Traceback (most recent call last):
>   File "mypy/semanal.py", line 7240, in accept
>   File "mypy/nodes.py", line 1345, in accept
>   File "mypy/semanal.py", line 3206, in visit_assignment_stmt
>   File "mypy/semanal.py", line 3733, in process_type_annotation
>   File "mypy/semanal.py", line 6894, in defer
> AssertionError: Must not defer during final iteration
> .../lib/python3.12/site-packages/django-stubs/http/request.pyi:56: : note: use --pdb to drop into pdb
> ```

This one looks like #2461. See also #2479 for another similar error with a different backtrace.

> I managed to run `dmypy` instead with the `master` branch, but unfortunately I get a new error:
> 
> ```
> $ dmypy run
>  dmypy run
> Restarting: plugins changed
> Daemon stopped
> Daemon started
> ...: error: INTERNAL ERROR -- Please try using mypy master on GitHub:
> https://mypy.readthedocs.io/en/stable/common_issues.html#using-a-development-mypy-build
> Please report a bug at https://github.com/python/mypy/issues
> version: 1.13.0
> ...
> Daemon crashed!
> Traceback (most recent call last):
>   ...
>   File ".../lib/python3.12/site-packages/mypy_django_plugin/transformers/choices.py", line 194, in transform_into_proper_attr_type
>     has_lazy_label = _has_lazy_label(node)
>                      ^^^^^^^^^^^^^^^^^^^^^
>   File ".../lib/python3.12/site-packages/mypy_django_plugin/transformers/choices.py", line 36, in _has_lazy_label
>     for member_name in node.enum_members:
>                        ^^^^^^^^^^^^^^^^^
> AttributeError: 'TypeInfo' object has no attribute 'enum_members'
> ```

Argh. This is unfortunate. `TypeInfo.enum_members` was introduced in mypy 1.14, but we have 1.13 as the minimum. This was used as part of the fix in #2646. It looks like it was added in https://github.com/python/mypy/commit/58f76280d816bfdc1ce625c0c0ab3519337bbbe7. We can probably extract the bit we need to keep supporting 1.13.

> Could you guys release a new version with this fix?

Once the above is fixed, then I guess we can ask @sobolevn for a release.

Sorry, my previous example was illustrative; here's an actual repro:

```python
~/e/django-stubs> git diff
diff --git a/tests/assert_type/db/models/test_enums.py b/tests/assert_type/db/models/test_enums.py
index 9634a960..9bf2bd98 100644
--- a/tests/assert_type/db/models/test_enums.py
+++ b/tests/assert_type/db/models/test_enums.py
@@ -92,6 +92,10 @@ class VoidChoices(BaseEmptyChoices):
     CHASM = enum.auto()


+class ChoiceWrapper:
+    Suit = Suit
+
+
 # Choice type that has been aliased to test that the plugin resolves types correctly.
 CompassPoint = imported.Direction

@@ -128,6 +132,7 @@ assert_type(Suit.names, list[str])
 assert_type(Suit.labels, list[_StrOrPromise])
 assert_type(Suit.values, list[int])
 assert_type(Suit.choices, list[tuple[int, _StrOrPromise]])
+assert_type(ChoiceWrapper.Suit.choices, list[tuple[int, _StrOrPromise]])
 assert_type(Suit.CLUB, Literal[Suit.CLUB])
 assert_type(Suit.CLUB.name, Literal["CLUB"])
 assert_type(Suit.CLUB.label, _StrOrPromise)

~/e/django-stubs> uv run mypy --strict tests/assert_type/db/models/test_enums.py
tests/assert_type/db/models/test_enums.py:135: error: Expression is of type "Any", not "List[Tuple[int, Union[str, _StrPromise]]]"  [assert-type]
tests/assert_type/db/models/test_enums.py:135: error: Unable to resolve type of choices property  [misc]
Found 2 errors in 1 file (checked 1 source file)
```

Thanks

--- a/mypy_django_plugin/transformers/choices.py
+++ b/mypy_django_plugin/transformers/choices.py
@@ -5,6 +5,7 @@ from mypy.types import (
     AnyType,
     Instance,
     LiteralType,
+    Overloaded,
     ProperType,
     TupleType,
     TypeOfAny,
@@ -184,6 +185,8 @@ def transform_into_proper_attr_type(ctx: AttributeContext) -> MypyType:
             _node_type = get_proper_type(_node_type.upper_bound)
         if isinstance(_node_type, Instance):
             node = _node_type.type
+        if isinstance(_node_type, Overloaded) and _node_type.is_type_obj():
+            node = _node_type.type_object()
 
     if node is None:
         if isinstance(_node_type, UnionType):

--- a/tests/assert_type/db/models/test_enums.py
+++ b/tests/assert_type/db/models/test_enums.py
@@ -1,7 +1,7 @@
 import enum
 from typing import Any, Literal, TypeVar
 
-from django.db.models import Choices, IntegerChoices, TextChoices
+from django.db.models import Choices, IntegerChoices, Model, TextChoices
 from django.utils.functional import _StrOrPromise
 from django.utils.translation import gettext_lazy as _
 from typing_extensions import assert_type
@@ -121,6 +121,18 @@ class ShoutyTextChoices(TextChoices):
     @property
     def label(self) -> str:
         return super().label.upper()
+
+
+class DeckModel(Model):
+    # Alias with the same name.
+    Suit = Suit
+
+    # Alias with a different name.
+    House = Suit
+
+    class Joker(TextChoices):
+        BLACK = "B"
+        RED = "R"
 
 
 # Assertions for an integer choices type that uses a lazy translatable string for all labels.
@@ -276,3 +288,9 @@ assert_type([member.value for choices in x for member in choices], list[int | st
 x = (Constants, Separator)
 assert_type([member.label for choices in x for member in choices], list[_StrOrPromise])
 assert_type([member.value for choices in x for member in choices], list[Any])
+
+
+# Assertions for choices objects defined and aliased in a model.
+assert_type(DeckModel.Suit.choices, list[tuple[int, _StrOrPromise]])
+assert_type(DeckModel.House.choices, list[tuple[int, _StrOrPromise]])
+assert_type(DeckModel.Joker.choices, list[tuple[str, str]])  # pyright: ignore[reportAssertTypeFailure]

